components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LoggingResource.java

```diff
@@ -18,23 +18,37 @@
 
 package org.wso2.micro.integrator.management.apis;
 
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.apache.commons.configuration.PropertiesConfigurationLayout;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
 import org.apache.log4j.Level;
-import org.apache.log4j.LogManager;
-import org.apache.log4j.Logger;
 import org.apache.synapse.MessageContext;
 import org.apache.synapse.commons.json.JsonUtil;
 import org.apache.synapse.core.axis2.Axis2MessageContext;
 import org.json.JSONObject;
+import org.wso2.carbon.utils.ServerConstants;
 
-import java.util.Enumeration;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
 import java.util.HashSet;
 import java.util.Objects;
 import java.util.Set;
 
 public class LoggingResource extends ApiResource {
 
+    private static final Log log = LogFactory.getLog(LoggingResource.class);
+
     private static final Level[] logLevels = new Level[]{Level.OFF, Level.TRACE, Level.DEBUG, Level.INFO, Level.WARN, Level.ERROR, Level.FATAL};
 
+    private String carbonHome = System.getProperty(ServerConstants.CARBON_HOME);
+    private String filePath = carbonHome + File.separator + "conf" + File.separator + "log4j2.properties";
```

Comment:
This will not work if we mount the conf directory separately. Can we use the system property, 'carbon.config.dir.path' instead?

Reply:
Fixed in 8d41bd6044fc5c95f39f6bf345f0876a1e3947d0